CMake Error at cmake/modules/LLVMExternalProjectUtils.cmake:86 (is_msvc_triple):

```cmake
llvm/cmake/modules/LLVMExternalProjectUtils.cmake:86
    is_msvc_triple(is_msvc_target ${target_triple})

But the definition has 
# is_msvc_triple(out_var triple)
#   Checks whether the passed triple refers to an MSVC environment
function(is_msvc_triple out_var triple)
```

I get error with this cmake config:
```sh
cmake -DCMAKE_BUILD_TYPE=Release -GNinja \
  ../llvm \
  -DCMAKE_C_COMPILER=/usr/bin/clang \
  -DCMAKE_CXX_COMPILER=/usr/bin/clang++ \
  -DLLVM_TARGETS_TO_BUILD:STRING='ARM;AArch64' \
  -DCAN_TARGET_i386:INTERNAL=False \
  -DCOMPILER_RT_DEFAULT_TARGET_ONLY=False \
  -DBUILD_SHARED_LIBS:BOOL=OFF \
  -DCLANGD_INCLUDE_TESTS:BOOL=OFF \
  -DLLVM_ENABLE_RUNTIMES="libcxx;libcxxabi" \
  -DLLVM_ENABLE_PROJECTS="clang;lld"
```



```
-- LLD version: 18.0.0
CMake Error at cmake/modules/LLVMExternalProjectUtils.cmake:86 (is_msvc_triple):
  is_msvc_triple Function invoked with incorrect arguments for function
  named: is_msvc_triple
Call Stack (most recent call first):
  runtimes/CMakeLists.txt:240 (llvm_ExternalProject_Add)
  runtimes/CMakeLists.txt:453 (runtime_default_target)
```

Steps to repro
1. Apply
```patch
+include(CMakePrintHelpers)
 # llvm_ExternalProject_BuildCmd(out_var target)
 #   Utility function for constructing command lines for external project targets
 function(llvm_ExternalProject_BuildCmd out_var target bin_dir)
@@ -79,9 +79,13 @@ function(llvm_ExternalProject_Add name source_dir)
 
   if(NOT ARG_TARGET_TRIPLE)
     set(target_triple ${LLVM_DEFAULT_TARGET_TRIPLE})
+    cmake_print_variables(LLVM_DEFAULT_TARGET_TRIPLE)
   else()
     set(target_triple ${ARG_TARGET_TRIPLE})
-  endif()
+    cmake_print_variables(ARG_TARGET_TRIPLE)
+    endif()
+
+  cmake_print_variables(target_triple)
```

2. use
```
cmake ../llvm -DCMAKE_BUILD_TYPE=Debug \
  -GNinja \
  -DLLVM_ENABLE_RUNTIMES="libcxx;libcxxabi" \
  -DLLVM_TARGETS_TO_BUILD:STRING='ARM;AArch64;RISCV' \
  -DLLVM_ENABLE_PROJECTS="clang;lld"
```

You'll find
```txt
-- LLVM_DEFAULT_TARGET_TRIPLE=""
-- target_triple=""
CMake Error at cmake/modules/LLVMExternalProjectUtils.cmake:90 (is_msvc_triple):
  is_msvc_triple Function invoked with incorrect arguments for function
  named: is_msvc_triple
```

--- a/llvm/cmake/modules/LLVMExternalProjectUtils.cmake
+++ b/llvm/cmake/modules/LLVMExternalProjectUtils.cmake
@@ -83,7 +83,7 @@ function(llvm_ExternalProject_Add name source_dir)
     set(target_triple ${ARG_TARGET_TRIPLE})
   endif()
 
-  is_msvc_triple(is_msvc_target ${target_triple})
+  is_msvc_triple(is_msvc_target "${target_triple}")
 
   if(NOT ARG_TOOLCHAIN_TOOLS)
     set(ARG_TOOLCHAIN_TOOLS clang)

